packages/synapse-sdk/src/telemetry/types.ts

```diff
@@ -0,0 +1,147 @@
+export type Environment = 'test' | 'development' | 'staging' | 'production'
+
+export interface TelemetryConfig {
+  /**
+   * Whether to enable telemetry.
+   *
+   * You can also control this via the environment variable `SYNAPSE_TELEMETRY_DISABLED` or the global variable `SYNAPSE_TELEMETRY_DISABLED`.
+   *
+   * This value will also be false if `NODE_ENV === 'test'`.
+   *
+   * @default true
+   */
+  enabled?: boolean
+  environment?: Environment // optional: deployment environment
+  /**
+   * The name of the application using the SDK.
+   * This is used to identify the application in the telemetry data.
+   * This is optional and can be set by the user via the synapse.telemetry.setContext() method.
+   * If not set, the SDK will use 'synapse-sdk' as the default app name.
+   */
+  appName?: string
+  tags?: Record<string, string> // optional: custom tags
+}
+
+interface BaseTelemetryEvent {
+  requestId: string
+  ts: string // UTC timestamp
+}
+
+export interface ErrorEvent extends BaseTelemetryEvent {
+  type: 'error'
+  name: string
+  message: string
+  stack?: string
+  operation?: OperationType // SDK operation context when error occurred
+}
+
+export interface HTTPEvent extends BaseTelemetryEvent {
+  type: 'http'
+  method: string
+  urlTemplate: string // sanitized URL (host + path, no query strings)
+  status?: number
+  ok?: boolean
+  durationMs: number
+  // Storage provider identification for filtering and debugging
+  spHostname: string
+  spPath: string
+  spOperation: string
+}
+
+export interface OperationEvent extends BaseTelemetryEvent {
+  type: 'operation'
+  operation: OperationType
+  params: Record<string, any> // allowlisted parameters
+  success: boolean
+  durationMs: number
+}
+
+export interface CustomEvent extends BaseTelemetryEvent {
+  type: 'custom'
+  name: string
+  data: Record<string, any>
+  level: 'info' | 'warning' | 'error'
+}
+
+export const OPERATIONS = {
```

Comment:
Is there a reason this is an object with key=value, rather than just an arrary?

Reply:
Is it to support "'custom.operation': `{string}.{string}`"

Reply:
Ive boiled this down to a simple operation string type instead of a key map.

Will reduce further